packages/flutter/lib/src/material/material.dart

```diff
@@ -305,16 +330,42 @@ class _MaterialState extends State<Material> with TickerProviderStateMixin {
   final GlobalKey _inkFeatureRenderer = GlobalKey(debugLabel: 'ink renderer');
 
   Color _getBackgroundColor(BuildContext context) {
-    if (widget.color != null)
-      return widget.color;
-    switch (widget.type) {
-      case MaterialType.canvas:
-        return Theme.of(context).canvasColor;
-      case MaterialType.card:
-        return Theme.of(context).cardColor;
-      default:
-        return null;
+    Color color = widget.color;
+    if (color == null) {
+      if (widget.type == MaterialType.canvas) {
+        color = Theme.of(context).canvasColor;
+      } else if (widget.type == MaterialType.card) {
+        color = Theme.of(context).cardColor;
+      }
     }
+
+    if (color != null &&
+        widget.darkThemeOverlay &&
+        Theme.of(context).brightness == Brightness.dark) {
+      color = _darkThemeOverlayColor(color, widget.elevation);
+    }
+    return color;
+  }
+
+  // Lighten a color based on elevation by applying a transparent white overlay.
+  Color _darkThemeOverlayColor(Color color, double elevation) {
+    if (elevation <= 0)
+      return color;
+
+    // Compute the opacity for the given elevation according to the spec:
+    // https://material.io/design/color/dark-theme.html#properties
+    final double opacity =
+          elevation <=  1.0 ? 0.05
+        : elevation <=  2.0 ? 0.07
+        : elevation <=  3.0 ? 0.08
+        : elevation <=  4.0 ? 0.09
+        : elevation <=  6.0 ? 0.11
+        : elevation <=  8.0 ? 0.12
+        : elevation <= 12.0 ? 0.14
+        : elevation <= 16.0 ? 0.15
+        : 0.16;
+    final Color whiteOverlay = Color.fromRGBO(255, 255, 255, opacity);
```

Comment:
This could be `Colors.white.withOpacity(opacity)`

Reply:
Yeah, I thought of that, but didn't want to include 'colors.dart' just for `white`.  Looking at it again, that does look more readable.